statement.go

```diff
@@ -49,6 +82,55 @@ func (s *Stmt) NumInput() int {
 	return int(paramCount)
 }
 
+// ParamName returns the name of the parameter at the given index (1-based).
+func (s *Stmt) ParamName(n int) (string, error) {
+	if s.closed {
+		return "", errClosedStmt
```

Comment:
Since these are helper functions, as in, they are not part of Go's sql driver interface, we might want to wrap their errors in an API error, like we do for `TypeInfo`.
```Go
func NewTypeInfo(t Type) (TypeInfo, error) {
	name, inMap := unsupportedTypeToStringMap[t]
	if inMap && t != TYPE_ANY {
		return nil, getError(errAPI, unsupportedTypeError(name))
	}
// ...
```

Reply:
Done, although I don't fully understand why we should wrap the error, as the true error will be suppressed by the `getError` function. This formatting with `%s` prevents us from using `errors.Is` in the tests.

Reply:
Ah, this is indeed a shortcoming of `getError`... It should probably use [`errors.Join`](https://cs.opensource.google/go/go/+/go1.23.4:src/errors/join.go;l=19). Thanks for pointing that out! Would you prefer we keep the original error (until this is addressed in `getError`)?

Reply:
Yes, `errors.Join` will solve the problem. I have reverted the error-wrapping changes.